### PR TITLE
docs: update vault-k8s to 0.9.0

### PR DESCRIPTION
**Pull request by jasonodonnell**
Updated for the newest release of Vault K8s: https://github.com/hashicorp/vault-k8s/blob/master/CHANGELOG.md. The version numbers in the Helm documentation are still valid until we release a new Helm chart.

### CODE REVIEW

File: website/content/docs/platform/k8s/injector/annotations.mdx [hunk 1]
```diff
@@ -163,9 +166,19 @@ Vault annotations change how the Vault Agent containers communicate with Vault.
 example, Vault's address, TLS certificates to use, client parameters such as timeouts,
 etc.
 
-- `vault.hashicorp.com/auth-path` - configures the auth path for the Kubernetes
+- `vault.hashicorp.com/auth-config` - configures additional parameters for the configured 
+  authentication method. The name of the config is any unique string after 
+  `vault.hashicorp.com/auth-config-`, such as `vault.hashicorp.com/auth-config-role-id-file-path`.
```
Comment (calvn) [inline website/content/docs/platform/k8s/injector/annotations.mdx:171]: Is the subset of auth methods small enough for this annotation-based config that we can simply list them here?

File: website/content/docs/platform/k8s/injector/index.mdx [hunk 2]
```diff
@@ -50,15 +50,15 @@ useful to customize Vault Agent beyond what the provided annotations offer.
 ### Authenticating with Vault
 
 The primary method of authentication with Vault when using the Vault Agent Injector
-is the service account attached to the pod. At this time, no other authentication
-method is supported by the controller.
+is the service account attached to the pod. Other authentication methods can be configured 
+using annotations.
```
Comment (calvn) [inline website/content/docs/platform/k8s/injector/index.mdx:54]: Asked above, but can we include a list here on the auth methods that can be configured this way in here (or above)?

**Review by calvn — APPROVED** [review]
Small suggestion/question, otherwise 👍

File: website/content/docs/platform/k8s/injector/annotations.mdx (hunk 1 above)
Comment (jasonodonnell) [reply website/content/docs/platform/k8s/injector/annotations.mdx:171]: It's all auto-auth methods. I'll just make a link to auto-auth doc.

File: website/content/docs/platform/k8s/injector/index.mdx (hunk 2 above)
Comment (jasonodonnell) [reply website/content/docs/platform/k8s/injector/index.mdx:54]: It's all auto-auth methods. I'll just make a link to auto-auth doc.

**Pull request merged**